standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -1877,11 +1889,20 @@ private MTable getMTable(String catName, String db, String table) {
       }
 
       if (projectionFields == null) {
-        mtables = (List<MTable>) query.execute(db, catName, lowered_tbl_names);
+        if(tablePattern == null) {
+          mtables = (List<MTable>) query.execute(db, catName, lowered_tbl_names);
+        }else{
+          mtables = (List<MTable>) query.execute(db, catName, tablePattern);
```

Comment:
Shouldn't there be 3 possiblities here?
tablePattern != null && !lowered_tbl_names.isEmpty() --> execute() needs 4 arguments
lowered_tbl_names.isEmpty() && tablePattern != null --> execute() needs 3 arguments
!lowered_tbl_names.isEmpty() && tablePattern == null --> execute() needs 3 arguments